### ISSUE TITLE
CVE-2022-0778 security vulnerability on .net aspnet core 6 runtime

### ISSUE THREAD

**Issue by dondeetan** [dotnet/aspnetcore#40746]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When running security scan on mcr.microsoft.com/dotnet/aspnet:6.0-alpine, we're now finding a new CVE vulnerability: 

Total: 2 (HIGH: 2, CRITICAL: 0)

+--------------+------------------+----------+-------------------+---------------+--------------------------------------+
|   LIBRARY    | VULNERABILITY ID | SEVERITY | INSTALLED VERSION | FIXED VERSION |                TITLE                 |
+--------------+------------------+----------+-------------------+---------------+--------------------------------------+
| libcrypto1.1 | CVE-2022-0778    | HIGH     | 1.1.1l-r0         | 1.1.1n-r0     | openssl: Infinite loop in            |
|              |                  |          |                   |               | BN_mod_sqrt() reachable              |
|              |                  |          |                   |               | when parsing certificates            |
|              |                  |          |                   |               | -->avd.aquasec.com/nvd/cve-2022-0778 |
+--------------+                  +          +                   +               +                                      +
| libssl1.1    |                  |          |                   |               |                                      |
|              |                  |          |                   |               |                                      |
|              |                  |          |                   |               |                                      |
|              |                  |          |                   |               |                                      |

Last container build image was 3/8 from: https://hub.docker.com/_/microsoft-dotnet-aspnet/?tab=description

Do we have an ETA for container update with the security fixes for the above CVE? Thank you!

### Expected Behavior

Security Scans Pass for asp.net core 6 runtime

### .NET Version

6.0

**Comment by blowdart** [comment]
.NET does not include OpenSSL, we use what is provided by the OS.

And in containers that means the base image hasn't been updated yet, this isn't just a .net problem, it's a wider container issue.

https://devblogs.microsoft.com/dotnet/staying-safe-with-dotnet-containers/ details what you need to know about containers and vulnerabilities. I specifically draw your attention to the section on [Linux Packages](https://devblogs.microsoft.com/dotnet/staying-safe-with-dotnet-containers/?msclkid=eefbf550a65211ec886beaffe9e6effe#linux-packages).

_To summarize, not us, we can't fix it because it's awaiting patching in the base OS. We don't have ETAs because those belong to whatever base OS you picked when you choose a container.

Tagging @richlander as this is his bag :)

**Comment by dondeetan** [comment]
Make sense and yes my fault for not adding the base image we're using. We did a work-around in the container run time docker file by just updating the packages

RUN apk update && apk add -u libcrypto1.1 && apk add -u libssl1.1

Btw, we're using dotnet/aspnet:6.0-alpine

Thank you

**Comment by richlander** [comment]
Looks like we're waiting on this: https://github.com/alpinelinux/docker-alpine/issues/243

Thanks for reaching out.

**Comment by jiasli** [comment]
Hi team, I noticed 3.15.1 has already fixed [CVE-2022-0778](https://security.alpinelinux.org/vuln/CVE-2022-0778):

https://alpinelinux.org/posts/Alpine-3.15.1-released.html

> This release includes a fix for openssl [CVE-2022-0778](https://security.alpinelinux.org/vuln/CVE-2022-0778).

**Comment by richlander** [comment]
Here you go.

```bash
docker run --rm -it mcr.microsoft.com/dotnet/runtime:6.0-alpine3.15 cat /etc/os-release
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.15.1
PRETTY_NAME="Alpine Linux v3.15"
HOME_URL="https://alpinelinux.org/"
BUG_REPORT_URL="https://bugs.alpinelinux.org/"
```

`apk list` tells us that this is the openssl version in that image:

```bash
libssl1.1-1.1.1n-r0 x86_64 {openssl} (OpenSSL) [installed]
```

Look good?
